lib/ansible/galaxy/collection/__init__.py

```diff
@@ -1762,6 +1759,18 @@ def _get_file_hash(b_path, filename):  # type: (bytes, str) -> str
         return _consume_file(fp)
 
 
+def remove_or_unlink(path: bytes):
+    """ Removes the symlink/directory/file if it exists. """
+    path = pathlib.Path(to_text(path))
+
+    if path.is_symlink():
+        path.unlink()
+    elif path.is_dir():
+        shutil.rmtree(path)
+    elif path.exists():
+        os.remove(path)
```

Comment:
It is a good practice to mark branches that are supposed to be unreachable. Modern type checkers, Python 3.11 and backports have a helper for this:
```suggestion
        os.remove(path)

    assert_never(path)
```
*(https://typing.readthedocs.io/en/latest/source/unreachable.html / https://docs.python.org/3/library/typing.html#typing.assert_never / https://stackoverflow.com/a/74535743/595220)*

This requires a fallback import from `typing_extensions` which our MyPy setup has pre-installed.